sources/packages/web/src/components/aest/students/EditCASSupplierInfoModal.vue

```diff
@@ -0,0 +1,177 @@
+
+export default defineComponent({
+  components: { ModalDialogBase, ErrorSummary },
+  props: {
+    allowedRole: {
+      type: String as PropType<Role>,
+      required: true,
+    },
+  },
+  setup() {
+    const SUPPLIER_NUMBER_INPUT_FIELD_NAME = "Supplier number";
+    const SUPPLIER_NUMBER_INPUT_FIELD_MAX_LENGTH = 30;
+    const SUPPLIER_NUMBER_PAD_NUMBER_OF_CHARS = 7;
+    const SUPPLIER_SITE_CODE_INPUT_FIELD_NAME = "Site code";
+    const SUPPLIER_SITE_CODE_INPUT_FIELD_MAX_LENGTH = 3;
+    const SUPPLIER_SITE_CODE_PAD_NUMBER_OF_CHARS = 3;
+    const PADDING_CHAR = "0";
+
+    const { checkOnlyDigitsRule, checkLengthRule, checkNullOrEmptyRule } =
+      useRules();
+    const { showDialog, showModal, resolvePromise, loading } = useModalDialog<
+      AddCASSupplierAPIInDTO | boolean
+    >();
+    const addCASSupplierForm = ref({} as VForm);
+    const formModel = reactive({} as AddCASSupplierAPIInDTO);
+
+    const submit = async () => {
+      const validationResult = await addCASSupplierForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      // Copying the payload, as reset is making the formModel properties null.
+      const payload = { ...formModel };
+      resolvePromise(payload);
+      addCASSupplierForm.value.reset();
+    };
+
+    const cancel = () => {
+      addCASSupplierForm.value.reset();
+      addCASSupplierForm.value.resetValidation();
+      resolvePromise(false);
+    };
+
+    const supplierNumberValidationRule = (value: string) => {
+      const checkNullOrEmptyRuleResult = checkNullOrEmptyRule(
+        value,
+        SUPPLIER_NUMBER_INPUT_FIELD_NAME,
+      );
+      if (checkNullOrEmptyRuleResult !== true) {
+        return checkNullOrEmptyRuleResult;
+      } else {
+        const checkLengthRuleResult = checkLengthRule(
+          value,
+          SUPPLIER_NUMBER_INPUT_FIELD_MAX_LENGTH,
+          SUPPLIER_NUMBER_INPUT_FIELD_NAME,
+        );
+        if (checkLengthRuleResult !== true) {
+          return checkLengthRuleResult;
+        } else {
+          {
+            return checkOnlyDigitsRule(value, SUPPLIER_NUMBER_INPUT_FIELD_NAME);
+          }
+        }
+      }
+    };
+
+    const supplierSiteCodeValidationRule = (value: string) => {
+      const checkNullOrEmptyRuleResult = checkNullOrEmptyRule(
+        value,
+        SUPPLIER_SITE_CODE_INPUT_FIELD_NAME,
+      );
+      if (checkNullOrEmptyRuleResult !== true) {
+        return checkNullOrEmptyRuleResult;
+      } else {
+        const checkLengthRuleResult = checkLengthRule(
+          value,
+          SUPPLIER_SITE_CODE_INPUT_FIELD_MAX_LENGTH,
+          SUPPLIER_SITE_CODE_INPUT_FIELD_NAME,
+        );
+        if (checkLengthRuleResult !== true) {
+          return checkLengthRuleResult;
+        } else {
+          {
+            return checkOnlyDigitsRule(
+              value,
+              SUPPLIER_SITE_CODE_INPUT_FIELD_NAME,
+            );
+          }
+        }
+      }
+    };
```

Comment:
The method can be refactored to execute early returns and reduce its complexity.
```ts
const supplierSiteCodeValidationRule = (value: string) => {
  const checkNullOrEmptyRuleResult = checkNullOrEmptyRule(
    value,
    SUPPLIER_SITE_CODE_INPUT_FIELD_NAME,
  );
  if (checkNullOrEmptyRuleResult !== true) {
    return checkNullOrEmptyRuleResult;
  }
  const checkLengthRuleResult = checkLengthRule(
    value,
    SUPPLIER_SITE_CODE_INPUT_FIELD_MAX_LENGTH,
    SUPPLIER_SITE_CODE_INPUT_FIELD_NAME,
  );
  if (checkLengthRuleResult !== true) {
    return checkLengthRuleResult;
  }
  return checkOnlyDigitsRule(value, SUPPLIER_SITE_CODE_INPUT_FIELD_NAME);
};
```